mono/mini/liveness.c

```diff
@@ -37,6 +37,62 @@
 static void mono_analyze_liveness2 (MonoCompile *cfg);
 #endif
 
+
+#define INLINE_SIZE 16
+
+typedef struct {
+	int capacity;
+	gpointer data [INLINE_SIZE];
+} MonoPtrSet;
+
+static void
+mono_ptrset_init (MonoPtrSet *set)
+{
+	set->capacity = 0;
+}
+
+static void
+mono_ptrset_destroy (MonoPtrSet *set)
+{
+	if (set->capacity > INLINE_SIZE)
+		g_hash_table_destroy (set->data [0]);
+}
+
+static void
+mono_ptrset_add (MonoPtrSet *set, gpointer val)
+{
+	//switch to hashtable
+	if (set->capacity == INLINE_SIZE) {
+		GHashTable *tmp = g_hash_table_new (NULL, NULL);
+		for (int i = 0; i < INLINE_SIZE; ++i)
+			g_hash_table_insert (tmp, set->data [i], set->data [i]);
+		set->data [0] = tmp;
```

Comment:
Would be nicer to store the hash as an additional field.

Reply:
That would waste memory thou